lib/src/realm_object.dart

```diff
@@ -352,7 +352,12 @@ extension RealmObjectInternal on RealmObject {
     return object;
   }
 
-  RealmObjectHandle get handle => _handle!;
+  RealmObjectHandle get handle {
+    final h = _handle;
+    if (h != null) return h;
+    throw RealmStateError('$this is not managed');
```

Comment:
this seems not in the correct place and most certainly would be an SDK bug.

Reply:
Previously this just did `=> _handle!` which would throw `_CastError:<Null check operator used on a null value>`. This can happen, if you try to index a managed list with an unmanaged object. I think that is a state error.

Reply:
If you try to index a managed list with an unmanaged object we should be checking that in the indexer itself. So the cast on a null value is something we should be seeing only while writing tests and new functionality.
So I think we should move this check for is object managed in the indexer.

Reply:
I agree with moving the check to the indexer, but we could probably keep the assertion here for a cleaner error message.

Reply:
This will mask the error for us and could hide from us in our tests since we expect the same error in some cases. 
The other error message `_CastError:<Null check operator used on a null value>` will make us fix it on the spot cause it will slip through any expects.

Reply:
I have replaced with an assert in debug. Also the `isManaged` check is done in `indexOf` now